core/templates/dev/head/pages/exploration_editor/editor_tab/state_editor_solution.html

```diff
@@ -0,0 +1,128 @@
+<div ng-controller="StateSolution">
+  <div ng-if="!isCurrentInteractionLinear() &&
+              stateHintsService.displayed.length > 0 &&
+              objectType !== 'LogicQuestion'">
+    <div class="oppia-editor-header">
+      <strong>Solution</strong>
+    </div>
+    <md-card class="oppia-editor-card-with-avatar">
+      <div class="oppia-editor-card-body">
+        <ul class="nav oppia-option-list nav-stacked nav-pills" role="tablist"
+            ng-if="solutionIsSpecified &&
+                   stateSolutionService.savedMemento.explanation">
+          <li ng-class="!isActive"
+              class="oppia-rule-block oppia-sortable-hint oppia-prevent-selection"
+              style="margin-top: 0;">
+            <a ng-click="toggleIsActive()"
+               class="oppia-rule-tab">
+              <div class="oppia-response-header-block">
+                <div class="oppia-response-header">
+                  <span angular-html-bind="getSolutionSummary()"></span>
+                  <span class="pull-right oppia-transition-200"
+                        ng-click="deleteSolution($event)"
+                        style="cursor: pointer; opacity: 0.5;">
+                    <i class="material-icons md-18">&#xE5CD;</i>
+                  </span>
+                </div>
+              </div>
+            </a>
+
+            <div ng-if="isActive">
+              <div class="oppia-editor-card-section">
+                <solution-editor solution="stateSolutionService.savedMemento"
+                                 object-type="objectType"
+                                 interaction-html="interactionHtml"
+                                 rule-description-choices="ruleDescriptionChoices"
+                                 on-save="onComponentSave">
+                </solution-editor>
+              </div>
+            </div>
+          </li>
+        </ul>
+      </div>
+    </md-card>
+
+    <md-card style="margin: 0; padding: 0;">
+      <div ng-if="editabilityService.isEditableOutsideTutorialMode()">
+        <button type="button"
+                class="btn btn-default btn-lg oppia-add-hint-button"
+                ng-click="openAddSolutionModal()">
+          + Add Solution
+        </button>
+      </div>
+    </md-card>
+  </div>
+</div>
+
+<script type="text/ng-template" id="modals/addSolution">
```

Comment:
Move this to a separate, named file. We are trying to move towards having just one "thing" per file.